src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -29,12 +29,19 @@ public static bool IsProcessRunning(int processId, string machineName)
         {
             // Performance optimization for the local machine:
             // First try to OpenProcess by id, if valid handle is returned verify that process is running
-            // Otherwise enumerate all processes and compare ids
-            if (!IsRemoteMachine(machineName))
+            // When the attempt to open a handle fails due to lack of permissions enumerate all processes and compare ids
+            // Attempt to open handle for Idle process (processId == 0) fails with ERROR_INVALID_PARAMETER
+            if (processId != 0 && !IsRemoteMachine(machineName))
             {
                 using (SafeProcessHandle processHandle = Interop.Kernel32.OpenProcess(ProcessOptions.PROCESS_QUERY_LIMITED_INFORMATION | ProcessOptions.SYNCHRONIZE, false, processId))
                 {
-                    if (!processHandle.IsInvalid)
+                    if (processHandle.IsInvalid)
+                    {
+                        int error = Marshal.GetLastWin32Error();
+                        if (error != Interop.Errors.ERROR_ACCESS_DENIED)
+                            return false;
```

Comment:
The `OpenProcess` method can fail for multiple reasons. I am not 100% sure that all errors except `ERROR_ACCESS_DENIED` mean that the process does not exist. However, I think that it's safe to assume that when the method fails with `ERROR_INVALID_PARAMETER` the process does not exist and we can safely return false (I've tested it locally).

```suggestion
                        if (error == Interop.Errors.ERROR_INVALID_PARAMETER)
                        {
                            Debug.Assert(processId != 0, "OpenProcess fails with ERROR_INVALID_PARAMETER for Idle Process");
                            return false;
                        }
```